sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py

```diff
@@ -387,26 +387,45 @@ def _to_generated(self) -> KeyValue:
         )
 
 
-class ConfigurationSettingsFilter:
+class ConfigurationSettingsFilter(Model):
     """Enables filtering of configuration settings."""
 
     key: str
     """Filters configuration settings by their key field. Required."""
     label: Optional[str]
     """Filters configuration settings by their label field."""
+    tags: Optional[List[str]]
+    """Filters key-values by their tags field."""
 
-    def __init__(self, *, key: str, label: Optional[str] = None) -> None:
+    _validation = {
+        "key": {"required": True},
+        "tags": {"unique": True},
+    }
+
+    _attribute_map = {
+        "key": {"key": "key", "type": "str"},
+        "label": {"key": "label", "type": "str"},
+        "tags": {"key": "tags", "type": "[str]"},
+    }
+
+    def __init__(
+        self, *, key: str, label: Optional[str] = None, tags: Optional[List[str]] = None, **kwargs: Any
+    ) -> None:
         """
         :keyword key: Filters configuration settings by their key field. Required.
         :paramtype key: str
         :keyword label: Filters configuration settings by their label field.
         :paramtype label: str or None
+        :keyword tags: Filters key-values by their tags field.
+        :paramtype tags: list[str]
         """
+        super().__init__(**kwargs)
         self.key = key
         self.label = label
+        self.tags = tags
 
 
-class ConfigurationSnapshot:  # pylint: disable=too-many-instance-attributes
+class ConfigurationSnapshot(Model):  # pylint: disable=too-many-instance-attributes
```

Comment:
Why add the `Model` baseclass? Aren't we using the `_to_generated` method?

Reply:
Yeah, we're using `_to_generated` method. `ConfigurationSetting` model also using the method, should we remove the `Model` base class of it?

Reply:
Update: removed base class in this class and `ConfigurationSettingsFilter`.